Add and Update Identity Providers (#839)
FusionAuth/fusionauth-site: site/docs/v1/tech/identity-providers/twitch.adoc

There is no further configuration to perform on this panel, you will only need to record or copy these values for the next step.

Once you have completed this configuration you will be able to enable the Twitch login button for one or more FusionAuth Applications. Below is an example login page with the Twitch Identity Provider enabled.

image::identity-providers/login-twitch.png[Twitch Login]

== Create a Twitch Identity Provider

:identity_provider: Twitch
include::docs/v1/tech/identity-providers/_idp_application_configuration.adoc[]

image::identity-providers/identity-provider-twitch-add.png[Twitch Home,width=1200]

:get_started_link: link:https://dev.twitch.tv/[Twitch - Getting Started]
:scope_addon: The values 'openid' 'user:read:email' are defaulted for this IdP.
:identity_provider: Twitch
:linking_strategy: Create a pending link

include::docs/v1/tech/identity-providers/_idp-form-fields.adoc[]

:linking_strategy!:
:identity_provider!:
:scope_addon!:
:get_started_link!:
+
+

> == Try a Login

Reviewer: This should be part of the table of contents.

> Twitch may ask on the login if you intend to redirect to your FusionAuth application.

Reviewer: ```suggestion
When a user logs in, Twitch may warn about the redirection to your FusionAuth application.
```

Does this happen every time? Is it only because we're in a development mode?

Author: Not entirely sure.  It's a single-time event until the cookies are cleared (I think).  I would actually have to do some more testing.

> image::identity-providers/twitch-warn-in-app.png[Twitch Warning, role=bottom-cropped]

Reviewer: Needs width

> Following, you may be presented with an overview page asking if grant permission to share your Twitch identity.

Reviewer: ```suggestion
After the user accepts the warning, they may be presented with an overview page asking if they grant permission to share their Twitch identity.
```

> image::identity-providers/twitch-overview.png[Twitch Overview]

Reviewer: Needs width and role of bottom-cropped.

> Once you click `Authorize`, Twitch will provide your users' information to FusionAuth.

Reviewer: ```suggestion
Once the user clicks `Authorize`, Twitch will provide the user's information to FusionAuth.
```

> FusionAuth give you three options to link this account:

Reviewer: Hmmm. This section seems weird. Twitch has a linking strategy, is this in addition to that? No other identity provider displays these three options, is Twitch different?

Author: @mooreds  

This was a "test" section of `Try A login` to show end-to-end what happens when a user logs in using the default behavior of FusionAuth and the corresponding IdP.  If we like it, I can create a separate PR to add it to other IdPs.  Part of me thinks that it may not be needed, but not sure.

1. `Create a Pending Link` is the default for Twitch.  
1. User Authenticates with Twitch
1. The user is redirected from Twitch back to FusionAuth.  
1. At this point, FusionAuth has a unique id from Twitch
1. FusionAuth then asks if they want to link to an existing FusionAuth user or if they want to make a new FusionAuth user.  
1. This can happen via the API but if using hosted pages, you get this image / workflow_page
![image](https://user-images.githubusercontent.com/16090626/128438319-0608b1a2-5ec0-4b7b-81b7-9ca6d2fbc673.png) 

I added a bit more text to clarify, but we can rework if needed.

Reviewer: I like this, but feel like it should be part of the linking docs, rather than the specific Identity Provider page. 

I think explaining more about how Identity Linking works is going to be helpful, because it's a new concept that provides a lot of options and flexibility, but is kinda confusing.

So my vote would be:

* remove this text/content from this PR (it is too big already :) ).
* add it to the Linking strategies section on the overview page. I'd add it to the examples subsection. (And make it clear that this is only handled when you use the hosted login pages.)
* Build out more examples for different strategies.
* Add 'linking third party identities' to the list of use cases that hosted login pages handle (here: https://fusionauth.io/docs/v1/tech/core-concepts/integration-points/#hosted-login-pages )

* `Link to an existing user` - This is a user already registered to your FusionAuth application
* `Create a new user` - This is a completely new user to your application
* `Cancel link request`

image::identity-providers/link-options-ui.png[Link Options, width=1600]

== Building Your Own Integration

:identity_provider: Twitch

> include::docs/v1/tech/identity-providers/_login_api_integration.adoc[]

Reviewer: Reset `identity_provider` below this line
